### PR TITLE
Fix: Autoformat regex in numbered lists. Closes #42.

### LINKED ISSUE #42: Numbers written in a paragraph are wrongly changed to a Numbered list after a blank space is added
Original bug: https://github.com/mozilla/notes/issues/429
Codepen Link for easy testing: https://codepen.io/vladikoff/pen/vWjGPe

I used the editor with GFM enabled from: https://github.com/szymonkups/ckeditor5-build-classic/tree/markdown_support

### STR

* Enable Markdown GFM plugin 
* Type a new line and type a number plus a space (example: `25 cakes`)

### Expected

The string should remain as `25 cakes`

### Actual

The 25 is converted to a `1.` (markdown numbered list)

### Details

I think Markdown numbered list should only activate if I type `1. ` (one plus a dot plus a space), but I might be wrong.

gfm2

*****

### PR DESCRIPTION
**Pull request by vladikoff**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Type: Fix regex to not skip . and ) in ordered lists. Closes #42.

---

### CODE REVIEW

File: src/autoformat.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ export default class Autoformat extends Plugin {
 
 		if ( commands.get( 'numberedList' ) ) {
 			// eslint-disable-next-line no-new
-			new BlockAutoformatEngine( this.editor, /^\d+[.|)]?\s$/, 'numberedList' );
+			new BlockAutoformatEngine( this.editor, /^\d+[.|)]\s$/, 'numberedList' );
```
Comment (szymonkups) [inline src/autoformat.js]: Do we want to start with any number or as @Reinmar [sugessted](https://github.com/ckeditor/ckeditor5-autoformat/issues/42#issuecomment-346466462) from `1`? I thought it was the main idea to support only `1.` and `1)` to trigger the list.

Comment (szymonkups) [reply src/autoformat.js]: But now I realized that maybe we should stay what you proposed.
What if someone is under the list and want to add another item. It is intuitive to write next number in order:
autoformat

Comment (Reinmar) [reply src/autoformat.js]: Good point, @szymonkups. Supporting only 1. would be unintuitive in some cases.

Comment (scofalik) [reply src/autoformat.js]: I am not sure if that's the way we want to go... What about having 9 list items and writing the tenth one? We can go like this forever.

Comment (Reinmar) [reply src/autoformat.js]: Yeah, I know. I hoped that no one will mention this :P

TBH, it just so much doesn't matter (after all it's just a helper and in 99% of cases people will start with `1.` or apply list manually if it didn't work) that all options are fine. So, let's just accept what we have and wait.

**Comment by vladikoff** [comment]
> Please update tests/manual/autoformat.md with the step about triggering numbered list.

Done!

**Comment by szymonkups** [comment]
Thanks, I will merge it with current master (there were some changes in the engine and repository was aligned to it). Then I will merge this back. Thanks @vladikoff.

**Pull request merged**